### PR TITLE
Feature/synk lagacy tests

### PR DESCRIPTION
**Pull request by VladislavKostyukevich**
# PR Details

Possibility to sync suites was added

## Related Issue

  - Add possibility to sync testsuites according to Not Executed results -> [View Issue](https://github.com/aquality-automation/aquality-tracking/issues/32)

## How Has This Been Tested

Import results with legacy tests
Open suite with legasy tests
Click Sync button
Select correct values
Synchronize suite
Check tests were removed from suite
Check only Not Executed results were removed
Check test run stats were updated

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/main/java/main/controllers/Project/SuiteController.java [hunk 1]
```diff
@@ -86,4 +90,60 @@ public boolean delete(TestSuiteDto template) throws AqualityException {
         }
         return testSuites;
     }
+
+    public void syncLegacyTests(List<TestDto> legacyTests, Integer suiteId, boolean removeNotExecutedResults) throws AqualityException {
+        if(baseUser.isManager() || baseUser.getProjectUserBySuiteId(suiteId).isManager()){
```
Comment (knysh) [inline src/main/java/main/controllers/Project/SuiteController.java]: Are you sure that sync should make only manager?
I think test results are responsibility of engineer.

File: src/main/java/main/controllers/Project/SuiteController.java [hunk 2]
```diff
@@ -86,4 +90,60 @@ public boolean delete(TestSuiteDto template) throws AqualityException {
...
+                    test2SuiteDao.delete(test2Suite);
+                    if(removeNotExecutedResults) {
+                        removePendingResultsForTest(suiteId, test.getId());
+                    }
+                } catch (AqualityException e) {
+                    e.printStackTrace();
+                }
+            });
+        }else{
+            throw new AqualityPermissionsException("Account is not allowed to Sync Test Suite", baseUser);
```
Comment (knysh) [inline src/main/java/main/controllers/Project/SuiteController.java]: Let's inform user what role necessary to use the feature

File: src/main/java/main/controllers/Project/SuiteController.java [hunk 3]
```diff
@@ -86,4 +90,60 @@ public boolean delete(TestSuiteDto template) throws AqualityException {
...
+                List<TestResultDto> testResults = testResultDao.searchAll(testResultTemplate);
+                tests = tests.stream().filter(test -> {
+                    TestResultDto currentResult = testResults.stream().filter(result -> result.getTest_id().equals(test.getId())).findFirst().orElse(null);
+                    return currentResult == null || currentResult.getFinal_result_id() == 3;
+                }).collect(Collectors.toList());
+            }
+
+            return tests;
+        } else {
+            throw new AqualityPermissionsException("Account is not allowed to Sync Test Suite", baseUser);
```
Comment (knysh) [inline src/main/java/main/controllers/Project/SuiteController.java]: Let's inform user what role necessary to use the feature

File: src/main/java/main/controllers/Project/SuiteController.java [hunk 4]
```diff
@@ -86,4 +90,60 @@ public boolean delete(TestSuiteDto template) throws AqualityException {
...
+            try {
+                testResultDao.delete(testResult);
+            } catch (AqualityException e) {
+                e.printStackTrace();
+            }
+        });
+    }
+
+    public List<TestDto> findLegacyTests(Integer suiteId, Integer notExecutedFor) throws AqualityException {
+        if (baseUser.isManager() || baseUser.getProjectUserBySuiteId(suiteId).isManager()) {
```
Comment (knysh) [inline src/main/java/main/controllers/Project/SuiteController.java]: same as for syncLegacyTests, are you sure that manager will do it?

File: src/main/java/main/controllers/Project/SuiteController.java (hunk 2 above)
Comment (VladislavKostyukevich) [reply src/main/java/main/controllers/Project/SuiteController.java]: Don't want do this cos user will never see it if he is not trying to brake the system without permissions

File: src/main/java/main/controllers/Project/SuiteController.java (hunk 4 above)
Comment (VladislavKostyukevich) [reply src/main/java/main/controllers/Project/SuiteController.java]: fixed

File: src/main/java/main/controllers/Project/SuiteController.java (hunk 3 above)
Comment (VladislavKostyukevich) [reply src/main/java/main/controllers/Project/SuiteController.java]: Don't want do this cos user will never see it if he is not trying to brake the system without permissions

File: src/main/java/main/controllers/Project/SuiteController.java (hunk 1 above)
Comment (VladislavKostyukevich) [reply src/main/java/main/controllers/Project/SuiteController.java]: fixed

**Pull request merged**